File: io-hotmoka-network/src/main/java/io/hotmoka/network/models/responses/MethodCallTransactionResponseModel.java
package io.hotmoka.network.models.responses;

import io.hotmoka.beans.responses.MethodCallTransactionResponse;

public abstract class MethodCallTransactionResponseModel extends CodeExecutionTransactionResponseModel {

	protected MethodCallTransactionResponseModel(MethodCallTransactionResponse response) {
        super(response);
    }

	protected MethodCallTransactionResponseModel() {}
}

File: io-hotmoka-network-thin-client/io-hotmoka-network-thin-client-kt/src/main/kotlin/io/hotmoka/network/thin/client/models/responses/MethodCallTransactionResponseModel.kt
package io.hotmoka.network.thin.client.models.responses

import io.hotmoka.network.thin.client.models.updates.UpdateModel

abstract class MethodCallTransactionResponseModel(
        updates: List<UpdateModel>,
        gasConsumedForCPU: String,
        gasConsumedForRAM: String,
        gasConsumedForStorage: String
) : CodeExecutionTransactionResponseModel(
        updates,
        gasConsumedForCPU,
        gasConsumedForRAM,
        gasConsumedForStorage
)